src/test/ui/consts/const-address-of.rs

```diff
@@ -0,0 +1,16 @@
+// check-pass
+
+#![feature(raw_ref_op)]
+
+const A: *const i32 = &raw const *&2;
+static B: () = { &raw const *&2; };
+static mut C: *const i32 = &raw const *&2;
```

Comment:
Could we test that `&raw const Cell::new(4)` fails somewhere? Either here or in a separate build-fail test?